### PR TITLE
Upgrade to AngularJS v1.6.6

### PR DESCRIPTION
**Pull request by marshmalien**
##### SUMMARY
This PR relates to https://github.com/ansible/awx/issues/280, and covers the upgrade from AngularJS 1.4.14 to the current release, AngularJS 1.6.6. It includes changes to packaging and other minor fixes related to the migration. 

##### ISSUE TYPE
 - Feature/Tech Debt Pull Request

##### COMPONENT NAME
- UI

##### AWX VERSION
``` awx: 1.0.1.5```

##### ADDITIONAL INFORMATION
These changes require you to run a `make clean-ui` and `npm install --prefix awx/ui/`. 

The AngularJS [migration guide](https://docs.angularjs.org/guide/migration) is a great reference to the the notable and breaking changes. 

* $location Service: Default hash-prefix for $location hash-bang URLs has changed from an empty string `('')` to a `('!')`. I added a config block to restore this back to the previous behavior. 
```
.config(['$locationProvider', function($locationProvider) {
   $locationProvider.hashPrefix('');
}])
```

* $http Service: Custom callback methods have been removed. Replace legacy `success()` and `error()` with the new Promise API `then()` and `catch()`. We have leveraged es6 destructuring to pull values out of the _response_ object. 

* input[type=radio]: Performs strict comparison (===) to determine its "checked" status. Had to convert the lookup modal values to match.

### CODE REVIEW

File: awx/ui/package.json [hunk 1]
```diff
@@ -120,7 +119,7 @@
     "lr-infinite-scroll": "git+https://git@github.com/lorenzofox3/lrInfiniteScroll",
     "moment": "~2.10.0",
     "ng-toast": "git+https://git@github.com/ansible/ngToast#2.0.1",
-    "nvd3": "git+https://git@github.com/ansible/nvd3#1.7.1",
+    "nvd3": "git+https://git@github.com/benthomasson/nvd3#1.7.1",
```
Comment (wenottingham) [inline awx/ui/package.json]: Can we just migrate this fork to the Ansible org?

**Comment by jaredevantabor** [comment]
Looks like the scheduler doesn't work with this version of Angular 
image

**Comment by gconsidine** [comment]
I noticed the vendor bundle size increased from 4.6 MB to 7.2 MB and I'm seeing  a couple warnings in the console along the lines of "attempted to load angular more than once" -- this could be related mismatched versions in the `angular-<whatever>` dependency projects, but I'm still investigating.

**Comment by mabashian** [comment]
Thing's i've noticed (some of this may not be related to the PR):

- Select Credential Type modal when creating a new cred -> clicking the row doesn't highlight the radio button (this seems to happen with all lookups(?))
- Sometimes after creating a resource the codemirror is off (noticed this after creating an inv and after creating a group but isn't consistent):
screen shot 2017-10-20 at 10 52 24 am

I've clicked around quite a bit and in general things seem to be functioning really well.

**Review by jaredevantabor — APPROVED** [review]
The tests are failing due to a flake8 error related to E722 which is causing the tests to fail on Shippable. Running flake8 locally on this branch passed however, as well as jshint. Because of this, this branch should be safe to merge. Nice work @marshmalien, this is really good to get merged in before Sprint 1.

**Pull request merged**